docs/ocs-on-a-disconnected-environment.md

```diff
@@ -53,129 +53,148 @@ you should get something similar to this:
+  quay.io/noobaa/pause@sha256:b31bfb4d0213f254d361e0079deaaebefa4f82ba7aa76ef82e90b4935ad5b105=<your_registry>/noobaa/pause
+  registry.redhat.io/ocs4/ocs-must-gather-rhel8@sha256:823e0fb90bb272997746eb4923463cef597cc74818cd9050f791b64df4f2c9b2
+  ```
 
 - **Mirror the images in `mapping.txt`**  
-```
-oc image mirror -f mapping.txt --insecure --registry-config=${AUTH_FILE}
-```
+  ```
+  oc image mirror -f mapping.txt --insecure --registry-config=${AUTH_FILE}
+  ```
+
+
+- **Add the images we mirrored in `mapping.txt` to the imageContentSourcePolicy**  
+After `oc adm catalog mirror` is completed it will print an output dir where an `imageContentSourcePolicy.yaml` is generated (usually the catalog image with `-manifests` suffix. e.g: `redhat-operators-manifests`).
+the missing related images should be added to the `ImageContentSourcePolicy` spec. e.g. an entry for rhceph-rhel8: 
+  ```yaml
+    - mirrors:
+      - <your_registry>/ocs4/rhceph-rhel8
+      source: registry.redhat.io/ocs4/rhceph-rhel8
+  ```
+
+
+- **After adding the related images (required for OCS-4.3 only), apply the `imageContentSourcePolicy.yaml` file to the cluster**  
+(the file is generated by `oc adm catalog mirror`. the output dir is usually `./[catalog image name]-manifests`)
+  ```
+  oc apply -f ./[output dir]/imageContentSourcePolicy.yaml
+  ```
+
+
+### lib-bucket-provisioner workarounds
+
+- OCS versions 4.3 and 4.4 are still dependent on lib-bucket-provisioner which is part of community-operators catalog. To work around this dependency it's necessary to build a catalog image for the comminity-operators. The dependency is only on OB\OBC CRDs which are installed with the CSV (which is included in the catalog image), so there is no need to mirror the catalog.
+
+  - **build oeprators catalog for community operators**  
+This step is required as workaround for lib-bucket-provisioner dependency.  
+(again, the tag of the `origin-operator-registry` in the `--from` should match the major and minor versions of the OCP cluster. e.g. 4.3)
+
+  ```
+  oc adm catalog build --insecure --appregistry-endpoint https://quay.io/cnr --appregistry-org community-operators --from=quay.io/openshift/origin-operator-registry:4.3 --to=${MIRROR_REGISTRY_DNS}/olm/community-operators:v1 --registry-config=${AUTH_FILE}
+  ```
+
+  - **Create CatalogSource for community-operators**  
+Create a CatalogSource object that references the catalog image for community-operators. Modify the following to your specifications and save it as a catalogsource.yaml file:
+  ```yaml
+  apiVersion: operators.coreos.com/v1alpha1
+  kind: CatalogSource
+  metadata:
+    name: community-operators-catalog
+    namespace: openshift-marketplace
+  spec:
+    sourceType: grpc
+    image: <your_registry>/olm/community-operators:v1 
+    displayName: Community Operators Catalog
+    publisher: grpc
```

Comment:
Publisher should be probably something else than `grpc`.

Reply:
was copied from the OLM docs. I'll change to OCS, we can change it later